src/probnum/linalg/solvers/beliefs/_linear_system_belief.py

```diff
@@ -0,0 +1,126 @@
+"""Linear system belief.
+
+Class defining a belief about the quantities of interest of a linear
+system such as its solution or the matrix inverse and any associated
+hyperparameters.
+"""
+
+from typing import Mapping, Optional
+
+from probnum import randvars
+
+try:
+    # functools.cached_property is only available in Python >=3.8
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
+
+# pylint: disable="invalid-name"
+
+
+class LinearSystemBelief:
```

Comment:
as discussed, we'll stick with one for now. The main reason being, that if we want to unify matrix / solution-based inference both x and Ainv should be in the base belief (even if only one of them is instantiated). But then in the exact case, beliefs over A and b can always be output by the algorithm's `solve` function, when it gets the `problem`. To then create a different belief object seems unnecessarily complicated. Instead of having `A` and `b` in the init one could think of having a `qoi` dictionary, which then can also hold other quantities of interest (like the belief over the residual, spectrum or else).